yjit/src/codegen.rs

```diff
@@ -3959,6 +3943,42 @@ fn gen_send_cfunc(
     asm.cmp(CFP, stack_limit);
     asm.jbe(counted_exit!(ocb, side_exit, send_se_cf_overflow).into());
 
+
+    if flags & VM_CALL_ARGS_SPLAT != 0 {
+        if cfunc_argc > 0 {
+            argc = pop_args_splat(argc, ctx, asm, ocb, side_exit, cfunc_argc as u32);
+        } else {
+            // This is a variadic c function and we'd need to pop
+            // each of the elements off the array, but the array may be dynmically sized
+            gen_counter_incr!(asm, send_args_splat_super);
+            return CantCompile
+        }
+    }
+
+    // pop_args_splat does a ctx.stack_pop so we can no longer side exit
+    let side_exit: Option<()> = None;
```

Comment:
You can still side-exit even if you do a `ctx.stack_pop()`, as long as you created your side-exit with a context prior to the stack pop.

Reply:
`ctx.stack_pop()` doesn't actually change the Ruby stack pointer, it moves a virtual stack pointer, so it should be possible to exit after a stack pop. Writing to the stack is a different thing though.

Reply:
Okay, I do push onto the stack. Will update my comment and the get_side_exit comment to only talk about push instead of push and pop.